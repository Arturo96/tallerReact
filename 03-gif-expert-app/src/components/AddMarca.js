import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddMarca = ({agregarMarca}) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    } 
    
    const handleSubmit = (e) => {
        e.preventDefault();

        console.log('Se llamÃ³ el handle Submit', inputValue);
        if(inputValue.trim().length > 2) {
            agregarMarca(inputValue);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input type="text" 
            value={inputValue}
            onChange={handleInputChange}
            />
        </form>
    )
}

AddMarca.propTypes = {
    agregarMarca: PropTypes.func.isRequired
}
