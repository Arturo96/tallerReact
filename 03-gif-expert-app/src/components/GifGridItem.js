import React from 'react';
import PropTypes from 'prop-types';


export const GifGridItem = ({title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} />
            <h4>{title}</h4>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}