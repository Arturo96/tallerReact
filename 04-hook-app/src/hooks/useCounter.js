import { useState } from "react"

export const useCounter = (initialState = 10) => {
    const [state, setState] = useState(initialState);

    const increment = (factor) => setState(state + factor);

    const reset = () => setState(initialState);

    const decrement = (factor) => setState(state - factor);
    
    return {
        state,
        increment,
        reset,
        decrement
    };
}
