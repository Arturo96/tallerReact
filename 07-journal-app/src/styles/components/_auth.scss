.auth {

    &__main {
        @include flexCenter();
        background-color: $primary;
        min-height: 100vh;
    }

    &__container {
        background-color: $bg-auth;
        max-width: 800px;
        width: 350px;
        padding: 1rem;
        border-radius: 5px;
        box-shadow: 0 3px 1px #000;
    }

    &__title {
        color: darken($primary, 20);
        text-align: center;
        @extend .mb-3;
    }

    &__input {
        background-color: inherit;
        border: none;
        border-bottom: 1px solid #ccc;
        color: $dark-grey;
        display: block;
        font-size: 1em;
        margin-bottom: 1rem;
        outline: none;
        padding: .35em;
        transition: border-bottom .2s;
        width: 100%;

        &:focus {
            border-bottom: 1px solid #333;
        }
    }

    &__social {
        @include flexCenter();
        flex-direction: column;
        padding: 1rem 0;
    }
    
    &__error {
        background-color: $error;
        border-radius: 5px;
        color: $bg-auth;
        @include flexCenter();
        margin-bottom: .5rem;
        padding: .5rem;
    }

    &__link {
        display: block;
        text-align: center;
        text-decoration: none;

        &:hover {
            text-decoration: underline;
        }
    }
}

