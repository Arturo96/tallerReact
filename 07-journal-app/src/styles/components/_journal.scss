.journal {

    &__main {
        position: relative;
        display: flex;
    }

    &__sidebar {
        background-color: $dark-grey;
        color: $white;
        position: fixed;
        display: flex;
        flex-direction: column;
        padding: 0 1rem;
        width: 30%;
        height: 100vh;

        &-nav {
            display: flex;
            align-items: center;
            justify-content: space-between;
        }

        h3 {
            font-weight: lighter;
        }
    }

    &__new-entry {
        @include flexCenter();
        cursor: pointer;
        flex-direction: column;
        margin-top: 2rem;
        transition: color .3s;

        &:hover {
            color: darken($white, 20);
        }
    }

    &__entries {
        flex: auto;
        margin-top: 2rem;
        overflow-y: auto;
    }

    &__entry {
        background-color: $white;
        border-radius: 4px;
        color: $dark-grey;
        cursor: pointer;
        display: flex;
        margin: 0 .5rem 1rem;
        margin-left: 0;
        overflow: hidden;
        transition: background-color .3s;

        &:hover {
            background-color: darken($white, $amount: 15);
        }

        &-picture {
            // height: $size-entry-picture;
            width: $w-picture;
            
        }

        &-body {
            width: calc(100% - #{$w-picture});
            padding: 1rem;
        }

        &-title {
            font-style: 14px;
            font-weight: bold;
        }

        &-content {
            margin-top: .5rem;
            font-size: 11px;
            line-height: 1.25;
        }

        &-date {
            @include flexCenter();
            flex-direction: column;
            padding: .5rem;

            span {
                font-size: .8em;
            }
        }
    }
}