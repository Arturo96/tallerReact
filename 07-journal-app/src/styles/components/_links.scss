.link {
    color: $dark-grey;
    text-decoration: none;
    display: block;

    &:hover {
        text-decoration: underline;
    }
}