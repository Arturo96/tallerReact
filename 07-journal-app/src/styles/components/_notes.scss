.notes {
	&__main {
		position: absolute;
		width: 70%;
		left: 30%;
		display: flex;
		min-height: 100vh;
		flex-direction: column;
		background-color: #EAF9D9;

	}

	&__appbar {
		display: flex;
		justify-content: space-between;
		align-items: center;

		background-color: $primary;
		color: $white;
		padding: 10px 20px;

		button {
			background-color: transparent;
			display: inline-block;
			transition: color 0.3s;

			&:hover {
				color: darken($white, 20);
			}
		}
	}

	&__content {
		display: flex;
		flex-direction: column;
		flex-grow: 1;
		padding: 1rem;

		& > * {
			margin-bottom: 1rem;
		}
	}

	&__input,
	&__textarea {
        border: none;
        outline: none;
	}

	&__input {
        color: $dark-grey;
		font-size: 1.25em;
		font-weight: 700;
	}

	&__textarea {
		border: 1px solid $black;
		font-size: 1.1em;
		flex-grow: 1;
		resize: none;
	}

	&__img {
		width: 30%;
		margin-bottom: 0;
		box-shadow: 5px 5px $dark-grey;
	}

	&__delete {
		width: 75%;
		margin: 1rem auto;
	}
}

.nothing {

	&__main {
		position: absolute;
		width: 70%;
		left: 30%;
	}
}
