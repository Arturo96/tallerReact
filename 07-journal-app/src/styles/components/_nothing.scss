.nothing {

    &__main {
        @include flexCenter();
        flex-direction: column;
        background-color: $primary;
        color: $white;
        min-height: 100vh;
    }
}