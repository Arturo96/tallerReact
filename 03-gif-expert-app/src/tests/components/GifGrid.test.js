import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import React from "react";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas del componente GifGrid", () => {
	const category = "Dragon Ball";

	test("Debe generarse correctamente", () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		});

		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs ", () => {
		const gifs = [
			{
				id: "ABC",
				title: "Random",
				url: "https://localhost/cosa.jpg"
			},
			{
				id: "CBDE",
				title: "Random",
				url: "https://localhost/cosa.jpg"
			}
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false
		});

		const wrapper = shallow(<GifGrid category={category} />);

		// expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("p").exists()).toBe(false);
		expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
	});
});
