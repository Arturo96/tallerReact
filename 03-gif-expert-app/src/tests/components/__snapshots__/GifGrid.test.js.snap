// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGrid Debe generarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__fadeIn"
  >
    Cargando...
  </p>
  <div
    className="cards"
  />
</Fragment>
`;
